python/jiminy_py/src/jiminy_py/simulator.py

```diff
@@ -230,37 +373,137 @@ def __dir__(self) -> List[str]:
         """
         return [*super().__dir__(), *dir(self.engine)]
 
+    def _create_viewer(self,
+                       robot: jiminy.Robot,
+                       robot_state: jiminy.RobotState,
+                       robot_color: Optional[
+                           Tuple[float, float, float, float]],
+                       **viewer_kwargs: Any) -> None:
+        """Helper method to create a viewer associated with a given robot.
+
+        :param robot: The robot of which the viewer will be created.
+        :param robot_state: The state associated to the robot.
+        :param color: the color the robot will be in the viewer.
+        :param viewer_kwargs: Extra keyword arguments to forward at `Viewer`
+                              initialization.
+        """
+
+        if len(self.robots) == 1:
+            robot_color = None
+
+        self._viewers.append(Viewer(
+            robot,
+            robot_name=re.sub('[^A-Za-z0-9_]', '_', robot.name),
+            use_theoretical_model=False,
+            open_gui_if_parent=False,
+            robot_color=robot_color,
+            **viewer_kwargs))
+
+        assert self._viewers[-1].backend is not None
+
+        # Share the external force buffer of the viewer with the engine
+        if self.is_simulation_running:
+            self._viewers[-1].f_external = [*robot_state.f_external][1:]
+        if self._viewers[-1].backend.startswith('panda3d'):
+            # Enable display of COM, DCM and contact markers by default if
+            # the robot has freeflyer.
+            if robot.has_freeflyer:
+                if "display_com" not in viewer_kwargs:
+                    self._viewers[-1].display_center_of_mass(True)
+                if "display_dcm" not in viewer_kwargs:
+                    self._viewers[-1].display_capture_point(True)
+                if "display_contacts" not in viewer_kwargs:
+                    self._viewers[-1].display_contact_forces(True)
+
+            # Enable display of external forces by default only for
+            # the joints having an external force registered to it.
+            if "display_f_external" not in viewer_kwargs:
+                profile_forces, *_ = self.engine.profile_forces.values()
+                force_frames = set(
+                    robot.pinocchio_model.frames[f.frame_index].parent
+                    for f in profile_forces)
+                impulse_forces, *_ = self.engine.impulse_forces.values()
+                force_frames |= set(
+                    robot.pinocchio_model.frames[f.frame_index].parent
+                    for f in impulse_forces)
+                visibility = self._viewers[-1]._display_f_external
+                assert isinstance(visibility, list)
+                for i in force_frames:
+                    visibility[i - 1] = True
+                self._viewers[-1].display_external_forces(visibility)
+
+    @property
+    def viewer(self) -> Optional[Viewer]:
```

Comment:
Change the return type in favor of "Viewer" and raise an exception if none is available.